File: frontend/src/entity/AppointmentDTO.ts
import {AppointmentStatus} from "./enums/AppointmentStatus";
import {AppointmentType} from "./enums/AppointmentType";

export interface AppointmentDTO {
    appointmentId:number;
    studentUsername:string;
    studentName:string;
    teacherUsername:string;
    teacherName:string;
    description:string;
    anonymous?: boolean;
    appointmentType?:AppointmentType;
    startTime:Date;
    endTime:Date;
    applyTime:Date;
    acceptTime?: Date | null;
    status:AppointmentStatus;
    rejectReason?: string | null;
    reschedulePending?: boolean;
    overdueFlagged?: boolean;
}

File: backend/src/main/java/com/ucaacp/backend/entity/DTO/AppointmentDTO.java
package com.ucaacp.backend.entity.DTO;

import com.ucaacp.backend.entity.enums.AppointmentStatus;
import com.ucaacp.backend.entity.enums.AppointmentType;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;


@Data
@NoArgsConstructor
@AllArgsConstructor
public class AppointmentDTO {
    private Integer appointmentId;
    private String studentUsername;
    private String studentName;
    private String teacherUsername;
    private String teacherName;
    private String description;
    private Boolean anonymous;
    private AppointmentType appointmentType;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime applyTime ;
    private LocalDateTime acceptTime;
    private AppointmentStatus status;
    private String rejectReason;
    private Boolean reschedulePending;
    private Boolean overdueFlagged;

}
